src/Terrain/Maps/MapReader.java

```diff
@@ -38,7 +38,7 @@
               continue;
             // underscores->spaces makes it pretty
             mapName = mapName.replaceAll("_", " ");
-            mapName = mapName.replaceAll(".map", "");
+            mapName = mapName.replaceAll("\\.map", "");
```

Comment:
If I have a cool map, called say "CoolMap.map", then the old line will render this in-game as "Coo", and the new line will render it as "CoolMap".
In regex, '.' is a wild-card. The slashes simply indicate that the '.' is to be treated as a normal character.
Two slashes are required because the Java interpreter also uses slashes as an escape character. So Java uses the first slash to escape the second, and passes "\.map" to the actual regex engine.

Reply:
I'm highly aware of how regexes work, but what I wasn't aware of is that the '.' doesn't apparently match spaces. This made it appear that the replacement string wasn't a regex.
The Europe Map is what made me figure there was no problem, heh.

Reply:
Silly me for explaining. I guess I figured that someone who is "highly aware of how regexes work" would have taken into account that 'm' and 'M' are not the same.
FYI, '.' *does* match spaces.

Reply:
I stand corrected. 
It appears I have been away from regexes for too long.
...and most of the text-searches I've been doing recently has been case-insensitive.
Also, your example was given as "CoolMap.map", which wouldn't match the lowercase 'm' anyway, which kind of invalidates your nit-picking, heh.

Reply:
Touché. That was a bad example on my part. Bummer.